src/lib.rs

```diff
@@ -311,3 +346,82 @@ impl Write for Device {
         Ok(())
     }
 }
+
+impl Device {
+    pub fn set_mpsse_clock(&mut self, freq: u32) -> std::result::Result<(), io::Error> {
+        const MAX: u32 = 30_000_000;
+        const MIN: u32 = 92;
+
+        assert!(
+            freq >= MIN,
+            "frequency of {} exceeds minimum of {}",
+            freq,
+            MIN
+        );
+        assert!(
+            freq <= MAX,
+            "frequency of {} exceeds maximum of {}",
+            freq,
+            MAX
+        );
+
+        let (divisor, clkdiv) = if freq <= 6_000_000 {
+            (6_000_000 / freq - 1, Some(true))
+        } else {
+            (30_000_000 / freq - 1, Some(false))
+        };
+
+        self.write_all(MpsseCmdBuilder::new().set_clock(divisor, clkdiv).as_slice())?;
+
+        Ok(())
+    }
+}
+
+impl MpsseCmdExecutor for Device {
+    type Error = io::Error;
+
+    fn init(&mut self, settings: &MpsseSettings) -> std::result::Result<(), io::Error> {
+        let millis = u8::try_from(settings.latency_timer.as_millis())
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+
+        if settings.reset {
+            self.usb_reset()
+                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        }
+
+        self.usb_purge_buffers()
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+
+        self.set_write_chunksize(settings.in_transfer_size);
+        self.set_read_chunksize(settings.in_transfer_size);
+        self.set_latency_timer(millis)
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        self.usb_set_event_char(0, false)
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        self.usb_set_error_char(0, false)
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        self.set_flow_control(FlowControl::RtsCts)
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        self.set_bitmode(0, BitMode::Reset)
```

Comment:
Is it known required to go through Reset, or just a precaution?

Reply:
This reset is added according to init sequence recommendations from AN135 "MPSSE Basics" by FTDI. For instance, see 4.2:
```
...
7) FT_SetBitMode – mode = 0, mask = 0 – Reset the MPSSE controller. Perform a general reset on
the MPSSE, not the port itself.
8) FT_SetBitMode – mode = 2, mask = 0 – Enable the MPSSE controller. Pin directions are set later
through the MPSSE commands.
```

Reply:
I see. Please add a comment referencing AN135 to this function (probably in the doc comment).

Reply:
Added doc comment with detailed init sequence and reference to AN 135.